syntax = "proto3";

package helloapi;
option java_multiple_files = true;

import "google/api/annotations.proto";

// https://cloud.google.com/endpoints/docs/grpc/transcoding
// curl -d '{"ping":"Music"}' http://localhost:8083/v1/hello/ping
service HelloService {

   rpc Health (EmptyRequest) returns (EmptyResponse) {
        option (google.api.http) = {
          get: "/health"
        };
    }


    rpc Echo (EchoRequest) returns (EchoResponse) {
        option (google.api.http) = {
          post: "/v1/hello/echo"
          body: "*"
        };
    }

    rpc Ping (PingRequest) returns (PingResponse) {
        option (google.api.http) = {
          post: "/v1/hello/ping"
          body: "*"
        };
    }
}

message EmptyRequest {
}

message EmptyResponse {
}

message EchoRequest {
    string name = 1;
}

message EchoResponse {
    string server_id = 1;
    string name = 2;
}

message PingRequest {
    string ping = 1;
}

message PingResponse {
    string ping = 1;
}